main_test.go

```diff
@@ -138,13 +140,33 @@ func TestScripts(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
+	cachePath, err := userCachePath()
+	if err != nil {
+		t.Fatal(err)
+	}
+
 	testscript.Run(t, testscript.Params{
 		Dir: filepath.Join("testdata", "scripts"),
 		Setup: func(e *testscript.Env) error {
 			e.Vars = append(e.Vars, "GOPROXY="+proxyURL)
 			e.Vars = append(e.Vars, "GONOSUMDB=*")
 			e.Vars = append(e.Vars, "HOME="+home)
+			e.Vars = append(e.Vars, "CACHEPATH="+cachePath)
```

Comment:
you already know HOME here, so you could make CACHEPATH an absolute dir. see https://godoc.org/github.com/rogpeppe/go-internal/testscript#Env. Then you don't need ` env CACHEDIR=$HOME/$CACHEPATH`.

Reply:
we actually don't use above HOME variable in test but created a new one with

> env HOME=$WORK/protoc_home

and as we don't know value $WORK until the test run, I think we couldn't construct CACHEPATH as an absolute directory.

Reply:
Hmm, you're right. I don't know who added the `env HOME=...`, but that's weird, because we already set up HOME here. You can clean that up here, or in another issue+PR; up to you.

We do know WORK here, though. See the testscript.Env link I pasted above.

Reply:
it's to prevent conflicts with the other test that downloads protoc (which also sets its own HOME): https://github.com/gunk/gunk/blob/d518a3cca47de41926d00552661c825816c2db41/testdata/scripts/download.txt#L5